scripts/imgtool/main.py

```diff
@@ -477,10 +487,46 @@ def sign(key, public_key_format, align, version, pad_sig, header_size,
         }
 
     img.create(key, public_key_format, enckey, dependencies, boot_record,
-               custom_tlvs, int(encrypt_keylen), clear, baked_signature,
+               custom_tlvs, compression_tlvs, int(encrypt_keylen), clear, baked_signature,
                pub_key, vector_to_sign)
-    img.save(outfile, hex_addr)
 
+    if re.match('.*\.bin$', infile) is not None :
```

Comment:
Use something like `os.path.splitext(infile)[-1].lower() == '.bin'` for better portability.